front/pages/manage_participants.js

```diff
@@ -0,0 +1,132 @@
+/*
+ * Manage the participants of event/citizen action
+ */
+export default() => {
+    const allCheckboxes = dom('#members-check-all');
+    const selectionCheckboxes = document.getElementsByName('members[]');
+    const contactButton = dom('#members-contact-button');
+    const contactSelection = dom('#members-contact-selection');
+    const exportButton = dom('#members-export-button');
+    const exportSelection = dom('#members-export-selection');
+    const printButton = dom('#members-print-button');
+    const printSelection = dom('#members-print-selection');
+
+    // Toggle action buttons
+    function toggleButtons(value) {
+        if (null != contactButton) contactButton.disabled = !value;
+        if (null != exportButton) exportButton.disabled = !value;
+        if (null != printButton) printButton.disabled = !value;
+    }
+
+    // Toggle each 'selection' checkbox
+    function toggleCheckboxs(value) {
+        const selectCheckboxes = (element) => {
+            element.checked = !!value;
+        };
+
+        selectionCheckboxes.forEach(selectCheckboxes);
+    }
+
+    // Bind the 'select all' checkbox
+    const bindAllCheckboxes = (event) => {
+        if (allCheckboxes.checked) {
+            toggleCheckboxs(true);
+            toggleButtons(true);
+        } else {
+            toggleCheckboxs(false);
+            toggleButtons(false);
+        }
+    };
+
+    on(allCheckboxes, 'change', bindAllCheckboxes);
+    allCheckboxes.dispatchEvent(new Event('change'));
+
+    const toggle = (event) => {
+        let all = true;
+
+        // Default behaviour
+        toggleButtons(false);
+        allCheckboxes.checked = false;
+
+        const toggleAllButtons = (elem) => {
+            if (elem.checked) {
+                // Enable button if at least one selection checkbox is checked
+                toggleButtons(true);
+            } else {
+                all = false;
+            }
+        };
+
+        selectionCheckboxes.forEach(toggleAllButtons);
+
+        // When all selection checkbox are checked, also check the 'select all' checkbox
+        if (all) {
+            allCheckboxes.checked = true;
+        }
+    };
+
+    const bindAllSelectionCheckboxes = (element) => {
+        on(element, 'change', toggle);
+        allCheckboxes.dispatchEvent(new Event('change'));
+    };
+
+    // Bind each 'selection' checkbox
+    selectionCheckboxes.forEach(bindAllSelectionCheckboxes);
+
+    const exportMembers = (event) => {
+        const members = [];
+        const addMember = (element) => {
```

Comment:
can you refacto this stuff for selecting the checked members to avoid the duplication with:
```
const contactMembers = (event) => {
 +        const members = [];
 +        const addMember = (element) => {
 +            if (element.checked) {
 +                members.push(element.value);
 +            }
 +        };
 +
 +        selectionCheckboxes.forEach(addMember);
 +
 +        contactSelection.value = JSON.stringify(members);
 +    };
```

and this 
```
 const printMembers = (event) => {
 +        const members = [];
 +        const addMember = (element) => {
 +            if (element.checked) {
 +                members.push(element.value);
 +            }
 +        };
 +
 +        selectionCheckboxes.forEach(addMember);
 +
 +        printSelection.value = JSON.stringify(members);
 +    };
```

:)